ecs-init/docker/docker.go

```diff
@@ -498,6 +547,37 @@ func getCapabilityExecBinds() []string {
 	return binds
 }
 
+// isDomainJoined is used to validate if container instance is part of a valid active directory.
+func isDomainJoined() bool {
+	realmPathByte, err := exec.Command("bash", "-c", "which realm").Output()
```

Comment:
exec.LookPath would be the proper way to do this: https://pkg.go.dev/os/exec#LookPath

Reply:
addressed suggestion - https://github.com/aws/amazon-ecs-agent/pull/3445 (as per meeting making this PR into smaller chunks)